### PR TITLE
(SIMP-3048) Update pupmod-simp-simp_elasticsearch to Elasticsearch 5.X

### PR DESCRIPTION
**Pull request by lnemsick-simp**
WORK ACCOMPLISHED:
- Eliminate sysconfig init defaults.
  - No longer need to add ES_USER and ES_GROUP.
  - ES_HEAP_SIZE is no longer supported as a sysconfig item,  or
    elasticsearch will fail to start if present.  Instead, we
    configure this in jvm.options
- Set heap size in jvm.options file, as ES_HEAP_SIZE environment
  variable is no longer supported by ES
- Set maximum number of threads via the nproc setting.
  Default value was too low and ES failed to start.
- Remove shards and replicas from node settings, as they
  must now be set per index.  ES failed to start with the
  old setting.
- Remove discovery.zen.ping.multicast.enabled node setting,
  since multicast support has been removed from ES.
- Updated repository for ES packages.  New repository is
  now at https://artifacts.elastic.co.
- Disabled ES use of sycall filter on EL6, as it is not
  supported by EL6
- Add following configuration parameters:
  $jvm_options
  $restart_config_change
  $restart_package_change
  $restart_plugin_change
  $jvm_options = undef
  $file_rolling_type
  $daily_rolling_date_pattern
  $rolling_file_max_backup_index
  $rolling_file_max_file_size
  $http_timeout
- Remove following configuration parameters:
  $plugintool - No longer supported by elastic/puppet-elasticsearch
  $max_log_days - No longer needed now that rolling log file support
    has been added to elastic/puppet-elasticsearch
  $https_trusted_nets - Not used
- Removed init_defaults, as is no longer needed
- Remove elasticsearch_log_purge cron job, as it is no longer needed.
- Change name of default ES instance to contain cluster name,
  instead of generic 'simp'.
- Used proxy_port parameter in elasticsearch.conf.erb, instead of hardcoding
  the port to 9199.
- Updated documentation
- Expanded tests

WORK NOT DONE:
- Restructure software to eliminate the circular relationship between
  parameters in init.pp and those in defaults.pp.
- Use the settings keystore for secure configuration (see
  https://www.elastic.co/guide/en/elasticsearch/reference/current/secure-settings.html)

### CODE REVIEW

**Comment by trevor-vaughan** [comment]
@lnemsick-simp Is this WIP? If so, could you edit the title to denote that please?

**Review by nick-markowski — DISMISSED** [review]

**Review by nick-markowski — DISMISSED** [review]

**Review by nick-markowski — DISMISSED** [review]

**Review by nick-markowski — DISMISSED** [review]
This still looks good to me

**Review by lnemsick-simp — COMMENTED** [review]
Found few problems on real SIMP system related to home directory for elasticsearch and noexec /tmp directory.  Need to 'puppetize' the solutions.

File: manifests/init.pp [hunk 1]
```diff
@@ -245,19 +275,40 @@
...
   }
 
-  pam::limits::rule { 'es_heap_sizelock':
-    domains => ['elasticsearch'],
-    type    => '-',
-    item    => 'memlock',
-    value   => 'unlimited',
-    order   => 0,
+  # Make sure elasticsearch user can spawn up to 2048 threads
+  if (versioncmp($facts['os']['release']['major'],'7') < 0) {
```
Comment (jeefberkey) [inline manifests/init.pp]: If using the major release version, the versioncmp function isn't needed.

File: manifests/init.pp [hunk 2]
```diff
@@ -227,10 +255,12 @@
     }
   }
 
+  $_http_listen_port = 9200
```
Comment (jeefberkey) [inline manifests/init.pp]: Should this be a parameter?

**Review by lnemsick-simp — COMMENTED** [review]
Integration testing has been successful.  Remaining work on Logstash filters and Grafana dashboards has been moved to a separate ticket.

**Review by trevor-vaughan — APPROVED** [review]
Code looks good and I've validated that the tests run properly under both Puppet 4.8 and 4.10.1.

**Pull request merged**